File: TPS/TP 1/paliRecursivo.java
import java.util.Scanner;

public class paliRecursivo{


    public static boolean verificar(String s, int inicio, int fim) {
        //condicao de parada, quando o indice "incio" for maior ou igual a 'fim', retorna true porquê nao houve ocorrencia de diferença dos caracteres
        if ( inicio >= fim) {
            return true;
        }

      
        if (s.charAt(inicio)  != s.charAt(fim)) {
            return false;
        }

        //chamada recursiva 
        return verificar(s, inicio + 1, fim - 1);
    }

public static void main (String[] args ){

Scanner sc = new Scanner (System.in);

String palavra;


palavra = sc.nextLine();

//enquanto a entrada for diferente de "FIM", verifica se é palindromoo
while(!(palavra.length()== 3 && palavra.charAt(0)=='F' && palavra.charAt(1)=='I')){

  
         if(verificar(palavra, 0, palavra.length()-1)){//agora, ao inves de passar somente a string, passa tambem o indice 0 e o tamanho menos 1, para percorrer das extremidades ate o meio
           System.out.println("SIM");
         }
         else{
         System.out.println("NAO");
         }
         palavra = sc.nextLine();
}


sc.close();

}
}

File: TPS/TP 1/paliRecursivo.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int verificar(char s[], int inicio, int fim) {
    
 //condicao de parada, quando o indice "incio" for maior ou igual a 'fim', retorna true porquê nao houve ocorrencia de diferença dos caracteres
    if ( inicio >= fim) {
        return 1;
        }

  
    if (s[inicio] != s[fim]) {
        return 0;
}

    //chamada recursiva 
    return verificar(s, inicio + 1, fim - 1);
}

    

int main(){

char palavra[1000];
scanf(" %[^\n]", palavra);

    //enquanto a entrada for diferente de "FIM", verifica se é palindromoo na função
    while(strcmp(palavra, "FIM")!=0){
             int tamanho = strlen(palavra)-1;

             if(verificar(palavra, 0, tamanho )){
                printf("SIM\n");
             }
             else{
                printf("NAO\n");
             }

             scanf(" %[^\n]", palavra);
    }

    return 0;
}